hal/explicit_pinmap.h

```diff
@@ -0,0 +1,183 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018-2019 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef EXPLICIT_PINMAP_H
+#define EXPLICIT_PINMAP_H
+#define CONSTEXPR constexpr
+#include "PeripheralPinMaps.h"
+
+constexpr PinMap get_pwm_pinmap_t(const PinName pin)
```

Comment:
These functions will need to use the `MSTD_CONSTEXPR_FN_14` macro from `<mstd_cstddef>` to not fall over with ARMC5. ARMC5 doesn't support C++14 constexpr, so these functions would not in fact be constexpr for it, and would fail to compile.

Obviously then these functions would not save ROM on ARMC5, as they'd pull in the runtime tables, but that just needs to be documented. Can't break compilation altogether.

Reply:
Fixed as suggested.